Makefile

```diff
@@ -50,13 +50,13 @@ _test: $(VENV)
 
 TEST_FILES = $(shell find tests -maxdepth 1 -name test_\*.py  | cut -c 7- | cut -d. -f1)
 $(TEST_FILES): $(VENV)
-	$(BIN)/pytest -k $@ $(ARGS)
+	. $(BIN)/activate && py.test -k $@ $(ARGS)
```

Comment:
why this change? shouldn't pytest in the virtualenv already "apply" that virtualenv? maybe it's just the dependency that needs expressing?

Reply:
It's for the venv's $PATH.

The functional tests executes talisker.gunicorn and friends as subprocesses, which if we run the test suite with the venv activated, are on the $PATH. Otherwise we needed some messy path munging to guess the executable path. It's also in line with how tests are run by tox (i.e. in an activated venv), so standardises it.

Reply:
:+1: thanks for the explanation